src/System/OS.php

```diff
@@ -0,0 +1,68 @@
+<?php
+
+namespace Dock\System;
+
+class OS
+{
+    const MAC = 1;
+    const LINUX = 2;
+    const WINDOWS = 3;
+
+    /**
+     * @return int Current Operating System
+     */
+    public function get()
+    {
+        $uname = strtolower(php_uname('s'));
+
+        switch ($uname) {
+            case 'darwin':
+                return self::MAC;
+            case 'linux':
+                return self::LINUX;
+            case 'win':
+                return self::WINDOWS;
+            default:
+                throw new \Exception("'$uname' is not a known operating system.");
+        }
+    }
+
+    public function createNewShell()
```

Comment:
I'm not sure about the responsibility of this method. Maybe it can be interesting to have a "shell creator/forwarder" interface with two implementations ?